### PR TITLE
Handle need to recreate and cancel

### PR DESCRIPTION
**Pull request by gbraad**
The user can cancel the preset with 'cancel' (and 'X') and will only ask the user to delete the preset when it differs from the current one.

### CODE REVIEW

File: src/preferences.ts [hunk 1]
```diff
@@ -157,6 +157,7 @@ async function configChanged(
   telemetryLogger: extensionApi.TelemetryLogger,
 ): Promise<void> {
   const currentConfig = await commander.configGet();
+  const currentStatus = await commander.status();
```
Comment (gbraad) [inline src/preferences.ts:160]: this makes it much slower...

File: src/preferences.ts [hunk 2]
```diff
@@ -306,11 +326,17 @@ async function handleRecreate(
...
+    return true;
   } else if (result === 'Delete') {
     await deleteCrc();
+    return true;
+  } else if (result === 'Cancel') {
+    // do nothing
+    return false;
   }
-}
+}
```
Comment (evidolob) [inline src/preferences.ts:342]: You deleted a new line on end of the file

**Comment by gbraad** [comment]
I tried this again and works
image

I set the preset to microshift on start. the bottom powershell shows this. I change it to openshift... but have to wait quite long before the dialog shows... I check the value and it removed microshift.
then when I 'cancel' (or 'X') it will default BACK to microshift as you cancelled. any other value will perform the delete as instructed.

File: src/preferences.ts (hunk 1 above)
Comment (gbraad) [reply src/preferences.ts]: No one suggested to use?

```typescript
if (crcStatus.status.Preset !== newConfig.preset) {
```

**Comment by evidolob** [comment]
Preset change still doesn’t work for me:

https://github.com/crc-org/crc-extension/assets/929743/fd187bd3-2ea2-42d1-8930-bbc4ed03d38b

**Comment by gbraad** [comment]
@evidolob you mentioned something else is the root cause of this?

**Review by evidolob — APPROVED** [review]

**Pull request merged**